arch/riscv/src/lib.rs

```diff
@@ -10,9 +10,9 @@
 
 pub mod csr;
 
-#[cfg(target_arch = "riscv32")]
+#[cfg(all(target_arch = "riscv32", not(doc)))]
 pub const XLEN: usize = 32;
-#[cfg(target_arch = "riscv64")]
+#[cfg(all(target_arch = "riscv64", not(doc)))]
```

Comment:
Usually we like to have these conditionals to be mutually exclusive, so that the item still exists in doc builds (but only one of them). So maybe change the 32-bit one to `any(target = ..., doc)`, which should always match exactly when the other definition is false?

Alternatively, which is much cleaner anyways, you might be able to collapse these into one expression by using something like `if cfg!(...) { 32 } else { 64 }`.

Reply:
Done! true this looks much cleaner.